### PR TITLE
OSBB posts crud on react for member

### PR DESCRIPTION
**Pull request by matsevich**
[Trello Task](https://trello.com/c/4qMhbmaz/87-osbb-posts-crud-on-react-for-member)

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [x] DB indexes added if needed
- [x] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [x] screenshot of feature is added if task has UI
Screenshot from 2020-12-20 13-05-59
Screenshot from 2020-12-20 13-06-12
Screenshot from 2020-12-20 13-12-52
Screenshot from 2020-12-20 13-56-31
Screenshot from 2020-12-20 13-57-10
Screenshot from 2020-12-20 13-57-16

### CODE REVIEW

**Review by mmarusyk — APPROVED** [review]

File: app/javascript/components/posts/PostsIndex/Post.module.scss [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  height: 150px;
+  margin-left: 20px;
+
+}
+
+.title {
+  text-align: left !important;
+}
+
+p {
```
Comment (mmarusyk) [inline app/javascript/components/posts/PostsIndex/Post.module.scss:30]: We can have problems with this style.  this style will be applied for all p. May create custom class

**Review by IvanDanyliv — APPROVED** [review]

**Review by Andrii-Kozak — APPROVED** [review]

File: app/javascript/components/posts/EditPost/EditPost.jsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  const handleSubmit = (values, long_description, image)=> {
+    const formData = new FormData();
+    formData.append('title', values.title);
+    formData.append('short_description', values.short_description);
+    formData.append('long_description', long_description);
+    formData.append('is_visible', values.is_visible);
+    if (image) formData.append('image', image) ;
+    const url = `/api/v1/posts/${props.match.params.id}`;
+    const csrfToken = document.querySelector('[name="csrf-token"]').content;
```
Comment (chu4) [inline app/javascript/components/posts/EditPost/EditPost.jsx:29]: same axios code as in CreatePost

File: config/initializers/elasticsearch.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
-if Rails.env == 'production'
-  url = ENV["BONSAI_URL"]
-  transport_options = { request: { timeout: 250 } }
-  options = { hosts: url, retry_on_failure: true, transport_options: transport_options }
-  Searchkick.client = Elasticsearch::Client.new(options)
+if Rails.env.production?
+    url = ENV["BONSAI_URL"]
```
Comment (chu4) [inline config/initializers/elasticsearch.rb:2]: too many tabs

File: db/migrate/20201124205435_create_posts.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+class CreatePosts < ActiveRecord::Migration[6.0]
+  def change
```
Comment (chu4) [inline db/migrate/20201124205435_create_posts.rb:2]: any indexes?

File: spec/controllers/api/v1/posts_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      end
+
+      it { is_expected.to have_http_status(:unprocessable_entity) }
+    end
+  end
+
+  describe 'PUT#update' do
+    context 'with valid params' do
+      before do
+        put :update, format: :json, params: { id: posts.id,
```
Comment (chu4) [inline spec/controllers/api/v1/posts_controller_spec.rb:56]:
```
       put :update, format: :json, params: {
         id: posts.id,
         post: valid_params.merge!(
           title: 'Example',
           short_description: 'Lorem',
           long_description: 'Sit amet',
           is_visible: true
         ) 
       }
```

**Review by Ukhanskyi — APPROVED** [review]

**Review by chu4 — APPROVED** [review]

**Pull request merged**